qubes-rpc/qubes-sync-clock

```diff
@@ -23,16 +23,42 @@ import re
 import subprocess
 
 def main():
-    stdin = sys.stdin.read(25)
-
-    untrusted_date_out = stdin.strip()
-
-    if not re.match(r'^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\+00:?00$', untrusted_date_out):
+    untrusted_date_out = sys.stdin.read(36)
```

Comment:
`sys.stdin.read()` returns `str`, not `bytes`. All the below handling of `bytes` (regex, comparing etc) are wrong.
If you really want bytes, then use `sys.stdin.buffer.read()`, but even then, save result of the decode call below and operate on str further down.

Reply:
I will fix QubesOS/qubes-core-admin#442 and then copy the changed code to here.